app/src/main/java/com/nextcloud/talk/chat/ChatActivity.kt

```diff
@@ -861,27 +861,17 @@ class ChatActivity :
                                 adapter?.notifyDataSetChanged()
                             }
 
-                            var lastAdapterId = 0
-                            if (adapter?.items?.size != 0) {
-                                val item = adapter?.items?.get(0)?.item
-                                if (item != null) {
-                                    lastAdapterId = (item as ChatMessage).jsonMessageId
+                            var lastAdapterId = getLastAdapterId()
+
+                            if (chatMessageList[0].jsonMessageId > lastAdapterId) { // Weird off by one bug
```

Comment:
this will destroy to fetch message from the past as processMessagesNotFromTheFuture won't be called when scrolling up.

Keep it like before will fix it:

```
     if (
          state.lookIntoFuture &&
          lastAdapterId != 0 &&
          chatMessageList[0].jsonMessageId > lastAdapterId
      ) {
          processMessagesFromTheFuture(chatMessageList)
      } else if (!state.lookIntoFuture) {
          processMessagesNotFromTheFuture(chatMessageList)
          collapseSystemMessages()
      }
```